js/huobi.js

```diff
@@ -3018,4 +3019,64 @@ module.exports = class huobi extends Exchange {
             }
         }
     }
+
+    async fetchFundingRateHistory (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        //
+        // Gets a history of funding rates with their timestamps
+        //  (param) symbol: Future currency pair
+        //  (param) limit: not used by huobi
+        //  (param) since: not used by huobi
+        //  (param) params: Object containing more params for the request
+        //  return: [{symbol, fundingRate, timestamp, dateTime}]
+        //
+        await this.loadMarkets ();
+        const request = {};
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchFundingRateHistory() requires a symbol argument');
+        } else {
+            const market = this.market (symbol);
+            request['contract_code'] = market['id'];
+        }
+        const response = await this.contractPublicGetLinearSwapApiV1SwapHistoricalFundingRate (this.extend (request, params));
+        //
```

Comment:
@Dan-krm To add functionality for coin-m perpetual swaps, you'll have to add this

```
let method = 'contractPublicGetLinearSwapApiV1SwapHistoricalFundingRate';
if (market['inverse']) {
    method = 'contractPublicGetSwapApiV1SwapHistoricalFundingRate';
}
const response = await this[method] (this.extend (request, params));
```

You'll also have to move the `const market = this.market (symbol);` line to outside the if statement and change `const` to `let`

Reply:
`base/quote:settle`

Symbols that look like `BTC/USDT:USDT` are linear (not inverse), because the quote currency matches the settle currency.
Symbols that look like `BTC/USDT:BTC` are inverse (not linear), because the base currency matches the settle currency.